### ISSUE TITLE
Reproducing Results from Eq. 12 and 13 of the Paper

### ISSUE THREAD

**Issue by katieluo88** [peiwang062/swag#4]
Hello, I was looking over the code and noticed the lines from `resnet_swag.py:115, 122, 128, 134` are commented out. I am a bit confused how the code reproduces the results from equations 12, 13 from the paper (where the softmax activation is applied to the loss function).

I'm wondering if I'm misunderstanding the paper? And if I wish to reproduce the results, should I use the function `softmax3d()` from line 189? Thanks in advance!

**Comment by peiwang062** [comment]
Hi, that part is just to give three different strategies. Any of them can reproduce the results. No huge difference, as explained in section 3.5 of the paper.
